articles/sso/v2/single-page-apps-sso.md

```diff
@@ -0,0 +1,177 @@
+
+You will then see the Lock screen, which allows you to provide the required credentials.
+
+![Lock screen](/media/articles/sso/v2/spa/lock.png)
+
+Once you've provided correct credentials, the app will indicate if you've successfully acquired a token.
+
+![Home page after logging in](/media/articles/sso/v2/spa/logged-in.png)
+
+If silent authentication succeeds, however, the app stores the token and its expiration date in your local storage.
+
+#### Auth0's Successful Authentication Response
+
+If the user is logged in via SSO already, Auth0 responds as if the user had manually authenticated using the SSO login page. You can extract the `access_token` from the hash fragment of the returned URL:
+
+```js
+function getParameterByName(name) {
+  var match = RegExp('[#&]' + name + '=([^&]*)').exec(window.location.hash);
+  return match && decodeURIComponent(match[1].replace(/\+/g, ' '));
+}
+
+function getAccessToken() {
+  return getParameterByName('access_token');
+}
+
+function getIdToken() {
+  return getParameterByName('id_token');
+}
+
+$(function () {
+  var access_token = getAccessToken();
+
+  // Optional: an id_token will be returned by Auth0
+  // if your response_type argument contained id_token
+  var id_token = getIdToken();
+
+  // Use the access token to make API calls
+  // ...
+});
+```
+
+The response for a call made *without* the `prompt=none` parameter is indistinguishable from the response for a call made *with* the parameter.
+
+You can test this using the sample app. If you're already logged in, you can request an updated token by clicking the **Click here to renew it** link.
+
+At this point, the app silently authenticates you, gets the new token, and updates the page to reflect your new token expiration datetime. Notice that you *did not* see the Lock screen asking for your credentials.
+
+#### Auth0's Error (Unsuccessful Authentication) Response
+
+If the user is not logged in via SSO, or if the SSO session has expired, Auth0 redirects to the specified `redirect_uri` and responds with an error. Auth0 returns error response parameters in a hash fragment.
+
+The following is a list of possible `ERROR_CODE` values, as defined by the [OpenID Connect specification](https://openid.net/specs/openid-connect-core-1_0.html#AuthError):
+
+* `login_required`: the user isn't logged in to Auth0, so Auth0 was unable to authenticate the user;
+* `consent_required`: the user is logged in to Auth0, but hasn't consented to authorizing the client app;
+* `interaction_required`: the user is logged in to Auth0, but needs to be redirected elsewhere before Auth0 can complete authentication (for example, if a [redirect rule](/rules/redirect) exists).
+
+If your user triggers any of these errors, your app must redirect the user to the Auth0 login page so that they can authenticate using a URL without the `prompt` parameter.
+
+## Refresh Expired Access Tokens
```

Comment:
I think there is too much duplication of content. Maybe remove this and add somewhere a note/panel with a link to the main silent authentication doc for details?